client/controllers/maps/eventMap.coffee

```diff
@@ -73,60 +74,73 @@ Template.eventMap.onRendered ->
             eventName: event.eventName
             lastIncidentDate: event.lastIncidentDate
             rgbColor: rgbColor
+            incidents: event.incidents
           eventIndex += 1
-    instance.templateEvents.set templateEvents
-    instance.disableNext.set if eventIndex < allEvents?.length then false else true
-    instance.disablePrev.set if currentPage is 0 then true else false
-    instance.subscribe('mapIncidents', _.pluck(templateEvents, '_id'))
+    @templateEvents.set templateEvents
+    @disableNext.set if eventIndex < allEvents?.length then false else true
+    @disablePrev.set if currentPage is 0 then true else false
 
   # Update the map markers to reflect user selection of events
-  @autorun ->
+  @autorun =>
     mapLocations = {}
-    templateEvents = instance.templateEvents.get()
-    Incidents.find(
-      userEventId: $in: instance.selectedEvents.find().map (e)-> e._id
-    ).map (incident)->
-      for location in incident.locations
-        latLng = location.latitude + "," + location.longitude
-        if not mapLocations[latLng]
-          mapLocations[latLng] =
-            name: location.name
-            incidents: []
-            events: []
-        mapLocations[latLng].events = _.union(
-          mapLocations[latLng].events, [incident.userEventId])
-        mapLocations[latLng].incidents = _.union(
-          mapLocations[latLng].incidents, [incident])
-
-    map.removeLayer(instance.mapMarkers)
-    markers = instance.mapMarkers = new L.FeatureGroup()
-    for coordinates, location of mapLocations
-      locationEvents = templateEvents.filter (e)->
-        e._id in location.events
-      popupHtml = Blaze.toHTMLWithData(Template.markerPopup, {
-        name: location.name
-        locationEvents: locationEvents.map (event)->
-          eventCopy = _.clone(event)
-          eventCopy.mostRecentIncident = _.chain(location.incidents)
-            .filter (i)-> i.userEventId == event._id
-            .sortBy (i)-> i.dateRange.start
-            .value()[0]
-          eventCopy
-      })
-      marker = L.marker(coordinates.split(","),
-        icon: L.divIcon
-          className: 'map-marker-container'
-          iconSize: null
-          html: MapHelpers.getMarkerHtml(locationEvents)
-      ).bindPopup(popupHtml, closeButton: false)
-      markers.addLayer(marker)
-    map.addLayer(markers)
-    if _.isEmpty(mapLocations)
-      map.setView([10, -0], 3)
-    else
-      map.fitBounds markers.getBounds(),
-        maxZoom: 10
-        padding: [20, 20]
+    templateEvents = @templateEvents.get()
+
+    # Create an array of the ids of the incidents of the selected events
+    incidentIds = []
+    @selectedEvents.find().fetch().forEach (event) ->
+      incidentIds = _.union(incidentIds, _.pluck(event.incidents, 'id'))
+    incidentIds
+
+    @subscribe 'mapIncidents', incidentIds, =>
```

Comment:
I think the way the subscription was done before was better because it didn't have to make a trip to the server every time an event is selected.